### PR TITLE
Port yuzu-emu/yuzu#3374: "input_common/udp: Minor changes"

### PR DESCRIPTION
**Pull request by FearlessTobi**
See yuzu-emu/yuzu#3374 for more details.

**Original description**:
Individual minor changes to the UDP backend that don't really warrant their own PRs by themselves.
Each commit should be fairly self-contained, so they can be reviewed individually.

---
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/5087)

### CODE REVIEW

File: src/input_common/udp/udp.cpp [hunk 1]
```diff
@@ -2,7 +2,9 @@
 // Licensed under GPLv2 or any later version
 // Refer to the license.txt file included.
 
-#include "common/logging/log.h"
+#include <mutex>
+#include <tuple>
+
```
Comment (zhaowenlan1779) [inline src/input_common/udp/udp.cpp]: I believe we do not separate includes.

**Comment by jroweboy** [comment]
The build errors are real, i think you are missing a few includes still
 
https://travis-ci.org/citra-emu/citra/jobs/645978070#L6398-L6456

File: src/input_common/udp/udp.h [hunk 2]
```diff
@@ -2,15 +2,14 @@
...
 #include <memory>
-#include <unordered_map>
-#include "input_common/main.h"
 #include "input_common/udp/client.h"
 
 namespace InputCommon::CemuhookUDP {
 
-class UDPTouchDevice;
-class UDPMotionDevice;
+class Client;
```
Comment (zhaowenlan1779) [inline src/input_common/udp/udp.h]: No need to forward declare anymore as this is already included.

Comment (lioncash) [reply src/input_common/udp/udp.h]: The include can be removed

Comment (FearlessTobi) [reply src/input_common/udp/udp.h]: Doing this only causes inclusion errors again.

Comment (lioncash) [reply src/input_common/udp/udp.h]: What are the inclusion errors?

Comment (zhaowenlan1779) [reply src/input_common/udp/udp.h]: @FearlessTobi I believe instead of fixing the errors here you can just include client.h where it is actually needed.

**Pull request merged**